### PR TITLE
NXBT-2950: Github oauth plugin

### PR DESCRIPTION
**Pull request by alexistimic**
Enable github oauth plugin and nexus realms
For now, 2 roles are mapped Github <=> Nexus : `nuxeo/Developers` && `nuxeo/Administrators`

### CODE REVIEW

File: base/supervisord-nexus-github-plugin.conf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[program:nexus-github-plugin]
+command=/bin/sh -xc "mkdir -p /opt/sonatype/nexus/system/com/larscheidschmitzhermes/ &&\
```
Comment (alexistimic) [inline base/supervisord-nexus-github-plugin.conf]: Following https://github.com/larscheid-schmitzhermes/nexus3-github-oauth-plugin

File: base/supervisord-nexus.conf [hunk 2]
```diff
@@ -1,5 +1,8 @@
 [program:nexus]
-command=/opt/sonatype/start-nexus-repository-manager.sh
+command=/bin/bash -xc "while [[ -z $(grep 'larscheidschmitzhermes' '/opt/sonatype/nexus/etc/karaf/startup.properties') ]];do \
```
Comment (alexistimic) [inline base/supervisord-nexus.conf]: supervisord always execute its target in parallel but we want `nexus` to start after the plugin have been deployed

File: parms/central/security-parms.json [hunk 3]
```diff
@@ -66,6 +74,14 @@
     "privileges": ["nx-all"],
     "roles": ["nx-admin", "nuxeo-developer"]
   },
+  {
```
Comment (alexistimic) [inline parms/central/security-parms.json]: Adjust Nexus `roleIDs` to match with Github `organization_name/team_name`

File: parms/central/security-parms.json [hunk 4]
```diff
@@ -50,6 +50,14 @@
     "privileges": ["All Repositories - (read)", "nx-component-upload", "nx-repository-view-*-*-*"],
     "roles": ["nuxeo-anonymous", "nx-developer"]
   },
+  {
+    "type":"role",
+    "id": "nuxeo/Developers",
+    "name": "nuxeo-Developers",
+    "description": "Github Developer role for Nuxeo",
+    "privileges": ["All Repositories - (read)", "nx-component-upload", "nx-repository-view-*-*-*"],
+    "roles": ["nuxeodev"]
```
Comment (jcarsique) [inline parms/central/security-parms.json]: "nuxeodev" ?

**Comment by jcarsique** [comment]
Pushed a few fixes to review please.
Also added user update method.

Related docs and code:
https://help.sonatype.com/repomanager3/security/privileges
https://help.sonatype.com/repomanager3/security/roles
https://help.sonatype.com/repomanager3/user-interface/uploading-components
https://github.com/sonatype/nexus-public/blob/master/components/nexus-security/src/main/java/org/sonatype/nexus/security/user/User.java
https://github.com/sonatype/nexus-public/blob/master/components/nexus-security/src/main/java/org/sonatype/nexus/security/SecurityApi.java
https://github.com/sonatype/nexus-public/blob/master/components/nexus-security/src/main/java/org/sonatype/nexus/security/SecuritySystem.java

**Pull request merged**